### PR TITLE
Support for timecode for vimeo and youtube videos.

### PR DESCRIPTION
**Pull request by kairamkondarajesh**
https://github.com/ymcatwincities/openy_gated_content/issues/142

## Steps to test:

- [ ] Pull the code from openy-142 branch.
- [ ] Then navigate to 'admin/content?type=gc_video (and look at this below screenshot).
image
- [ ] Edit any record from list of video and add fragment (eg: #t=30s) then save the node (and look at this below screenshot).
image
- [ ] Then navigate to frontend of the video page and observe the video starts from the mentioned time(#t-30s) (and look at this below screenshot).
image

### CODE REVIEW

**Comment by anpolimus** [comment]
@kairamkondarajesh 
Hello. Conpiled js file is missing.

**Comment by NightHunterSV** [comment]
Seems the code is not working good :(
After seeking the needed time, the video left "loading":
image
If we wouldn't be able to find a solution for this, we shouldn't merge this code.

**Comment by sarah-halby** [comment]
@kairamkondarajesh have you reviewed @NightHunterSV comment above? It's unclear to me who owns the next steps on this PR.

**Comment by kairamkondarajesh** [comment]
> Seems the code is not working good :(
> After seeking the needed time, the video left "loading":
> image
> If we wouldn't be able to find a solution for this, we shouldn't merge this code.
Sorry for late reply, Its working for me, Video is playing from mentioned time frame.

Screenshot 2021-07-15 230852
Screenshot 2021-07-15 230959

Can you please give me the  YouTube link, so that I will check in my side.

**Comment by sarah-halby** [comment]
@NightHunterSV see request above ^^

File: js/gated-content/src/components/MediaPlayer.vue [hunk 1]
```diff
@@ -6,12 +6,11 @@
       :videoId="videoId"
       :options="{responsive: 'true', url: media.field_media_video_embed_field}"
       :player-vars="handleAttributes()"
-      :autoplay="autoplay"
-      @loaded="handleLoaded()"
-      @ready="handleReady()"
+      @loaded="setTimeCode()"
```
Comment (NightHunterSV) [inline js/gated-content/src/components/MediaPlayer.vue:9]: You shouldn't just change existing callbacks to new ones. Please, put your code into existing methods: handleLoaded and handleReady. Also, keep in mind that we should autoplay videos only if the appropriate setting is enabled.

**Review by NightHunterSV — CHANGES_REQUESTED** [review]

**Comment by NightHunterSV** [comment]
As an example of the YouTube video -- you can use any video from the demo content, coming in the project:
modules/openy_gc_demo/migrations/migrate_plus.migration.virtual_y_media_video.yml

File: js/gated-content/src/components/MediaPlayer.vue (hunk 1 above)
Comment (kairamkondarajesh) [reply js/gated-content/src/components/MediaPlayer.vue]: We have updated the code. Can you please review once.

**Comment by kairamkondarajesh** [comment]
> As an example of the YouTube video -- you can use any video from the demo content, coming in the project:
> modules/openy_gc_demo/migrations/migrate_plus.migration.virtual_y_media_video.yml

We have tested with youtube link, its working fine.

**Comment by NightHunterSV** [comment]
1. Please, first merge the `master` branch into your code, or rebase your branch onto it. Cause now the changes are just removing everything that was done to handle the events logging, and so on.
2. `playerReadied` is never called since it is not used as a callback for the player event.
3. `setTimeCode` is also never used.

File: js/gated-content/src/components/MediaPlayer.vue [hunk 2]
```diff
@@ -113,6 +125,27 @@ export default {
...
+          const timecodeInSeconds = parseInt(timecode, 10);
+          this.$refs.player.player.setCurrentTime(timecodeInSeconds);
+        }
+      } else {
+        const timecode = this.media.field_media_video_embed_field.substring(
+          this.media.field_media_video_embed_field.lastIndexOf('#t=') + 3,
+          this.media.field_media_video_embed_field.lastIndexOf('s')
+        );
+        if (!isNaN(timecode)) {
+          const timecodeInSeconds = parseInt(timecode, 10);
```
Comment (NightHunterSV) [inline js/gated-content/src/components/MediaPlayer.vue:144]: Any reason this is done inside each section of the `if`?
Probably we should do just the last lines (player seek to the needed timecode) depending on which player is rendered?

**Comment by froboy** [comment]
I think this is fixed, but it just needs to have the JS recompiled. @kairamkondarajesh can you please one more time. :)

**Comment by NightHunterSV** [comment]
I'm afraid, we need improvements here.
This is what I have on the build when there is no timecode for the video set:

image

**Comment by NightHunterSV** [comment]
And even worse for the Vimeo video source:

image

**Review by NightHunterSV — APPROVED** [review]

**Review by hamrant — APPROVED** [review]

File: js/gated-content/src/components/MediaPlayer.vue [hunk 3]
```diff
@@ -48,20 +50,43 @@ export default {
...
+      ) {
+        return this.mediaUrl;
       }
       return this.media.field_media_video_id;
     },
+    mediaUrl() {
+      return this.media.field_media_video_embed_field;
+    },
+    timecode() {
+      const matches = this.mediaUrl.match(/[&?#]t=((\d+)h)?((\d+)m)?(\d+)s?/);
```
Comment (NightHunterSV) [inline js/gated-content/src/components/MediaPlayer.vue:75]: The details about using such regular expression:

image

image

**Review by anpolimus — APPROVED** [review]
Code is ok. Lets move to QA.

**Comment by sarah-halby** [comment]
@anpolimus & @kairamkondarajesh - let's make sure this is merged to the 1.6.1 point release since we are in code freeze for the 1.6 release on Nov 9.

**Comment by AnastasiiaPys** [comment]
retest

**Comment by AnastasiiaPys** [comment]
Tested.
Works as expected
image

**Review by AnastasiiaPys — APPROVED** [review]

**Pull request merged**